notebooks/Users/stevendleung@ischool.berkeley.edu/Airline Delays- Random Forest.py

```diff
@@ -48,97 +48,157 @@
 
-train_RF = stringIndexer.fit(train_RF).transform(train_RF)
 
-validation_RF = stringIndexer.fit(validation_RF).transform(validation_RF)
 
-# test_GBT = stringIndexer.fit(test_GBT).transform(test_GBT)
+# COMMAND ----------
+
+#INVESTIGATE % NULL
+from pyspark.sql.functions import isnan, when, count, col
+train_subset = train_set.select("origin_num_flights","origin_avg_dep_delay", "origin_pct_dep_del15", "origin_avg_taxi_time", "origin_avg_weather_delay", "origin_avg_nas_delay", "origin_avg_security_delay", "origin_avg_late_aircraft_delay", "dest_num_flights","dest_avg_dep_delay", "dest_pct_dep_del15", "dest_avg_taxi_time", "dest_avg_weather_delay", "dest_avg_nas_delay", "dest_avg_security_delay", "dest_avg_late_aircraft_delay", "carrier_num_flights", "carrier_avg_dep_delay", "carrier_avg_carrier_delay", "origin_WND_speed_rate", "origin_CIG_ceiling_height", "origin_VIS_distance", "origin_TMP_air_temperature", "origin_DEW_dew_point_temp", "dest_WND_speed_rate", "dest_CIG_ceiling_height", "dest_VIS_distance", "dest_TMP_air_temperature", "dest_DEW_dew_point_temp",
+"origin_aa1_rain_depth", "dest_aa1_rain_depth", "origin_aj1_snow_depth", "dest_aj1_snow_depth")
+display(train_subset.select([((count(when(col(c).isNull(), c)))/17247986).alias(c) for c in train_subset.columns]))
 
 # COMMAND ----------
 
-# Assemble features into vector
-features = categorical_index + numeric
-assembler = VectorAssembler(inputCols=features, outputCol="features")
+# # Assemble features into vector
+features = categorical + numeric
+assembler = VectorAssembler(inputCols=features, outputCol="features2")
 assembler.setHandleInvalid("skip")
 
-train_RF = assembler.transform(train_RF)
-validation_RF = assembler.transform(validation_RF)
-# test_GBT = assembler.transform(test_GBT)
+train_RF = assembler.transform(train_set)
+train_RF_OS = assembler.transform(train_set_oversampled)
+validation_RF = assembler.transform(val_set)
+# test_RF = assembler.transform(test_RF)
 
 # COMMAND ----------
 
-# Define RF model
-rf = RF(labelCol='label', featuresCol='features',numTrees=20)
-fit = rf.fit(train_RF)
+#DROP DEP DEL15 = null
+train_RF_drop_nulls = train_RF.where(f.col('dep_del15').isNotNull())
+train_RF_OS_drop_nulls = train_RF_OS.where(f.col('dep_del15').isNotNull())
+validation_RF_drop_nulls = validation_RF.where(f.col('dep_del15').isNotNull())
 
 # COMMAND ----------
 
-val_df = assembler_features.transform(val_set_dropna_output)
-val_df = val_df.withColumn("label", val_set_dropna_output.dep_del15)
+#Define rf model, Evaluator and paramgrid to use in CV
+rf = RF(labelCol='label', featuresCol='features2', numTrees = 30, maxDepth = 10)
 
-# COMMAND ----------
+evaluator = BinaryClassificationEvaluator(labelCol="label")
 
 paramGrid = ParamGridBuilder().build()
+#       .addGrid(rf.numTrees, [int(x) for x in np.linspace(start =20, stop = 30, num = 2)]) \
```

Comment:
Not something we need to change now, but if you haven't already we may want to up the number of samples to 3-5. Especially for maxDepth it could be that 5 is too small and 15 is too big so we would want to check somewhere in between too I think.